martin/src/srv/server.rs

```diff
@@ -127,15 +127,32 @@ pub fn new_server(config: SrvConfig, state: ServerState) -> MartinResult<(Server
             .wrap(middleware::NormalizePath::new(TrailingSlash::MergeOnly))
             .wrap(middleware::Logger::default())
             .configure(router)
-    })
-    .bind(listen_addresses.clone())
-    .map_err(|e| BindingError(e, listen_addresses.clone()))?
-    .keep_alive(keep_alive)
-    .shutdown_timeout(0)
-    .workers(worker_processes)
-    .run();
-
-    Ok((server, listen_addresses))
+    };
+
+    match () {
+        #[cfg(feature = "lambda")]
+        () if is_running_on_lambda() => {
```

Comment:
why not a simple block like this?  Why do you need a match statement?

```
#[cfg(feature = "lambda")]
if is_running_on_lambda() {
    ....
}
```

Reply:
P.S. we need to document that a new env var variable has magical properties per https://github.com/hanabu/lambda-web/blob/ddceea3b8ffcfae465fc8d0059aaa52a8ff94f8e/src/lib.rs#L34 

if `AWS_LAMBDA_RUNTIME_API` is defined, runs under AWS lambda runtime

Reply:
I think I was frustrated that an if/else wouldn't work with the conditional compilation, and this is a way to express that only one of the code paths will run. But an if statement and a return would probably be clearer.

Reply:
I may refactor it later, no worries. I think we should focus on the main issue - Testing, and ideally - testing in CI.  We need a clear way to ensure that after some person adds some random code, we don't have to do too much work to make sure it continues to work in all expected scenarios.